ruby/parser.js

```diff
@@ -20,14 +20,23 @@
     if (cs.layoutRegions) options += layoutRegions(cs.layoutRegions)
     if (cs.ignoreRegions) options += ignoreRegions(cs.ignoreRegions);
     if (cs.floatingRegions) options += floatingRegions(cs.floatingRegions);
-    if (cs.scrollRootElement ) throw new Error("Scroll root option not implemented in the ruby SDK")
+    if (cs.scrollRootElement )  element += scrollRootElement(cs.scrollRootElement)
     if (cs.ignoreDisplacements !== undefined) options += `.ignore_displacements(${cs.ignoreDisplacements})`
     if (cs.sendDom !== undefined) options += `.send_dom(${serialize(cs.sendDom)})`
+    if (cs.layoutBreakpoints ) options += `.layout_breakpoints(${serialize(cs.layoutBreakpoints)})`
     if (cs.variationGroupId) options += `.variation_group_id(${serialize(cs.variationGroupId)})`
-    if (cs.isFully) options += '.fully';
+    if (cs.isFully !== undefined) options += `.fully(${serialize(cs.isFully)})`;
+    if (cs.visualGridOptions) options += `.visual_grid_options(polyfillAdoptedStyleSheets: ${cs.visualGridOptions.polyfillAdoptedStyleSheets})`
+    if (cs.hooks) options += hooks(cs.hooks);
     if (cs.name) name = `'${cs.name}', `;
+    if (cs.matchLevel) options += `.match_level(${serialize(cs.matchLevel)})`
     return name + ruby + element + options
 
+    function hooks(obj){
+        const hooks = Object.keys(obj).map(key => `${key}: ${serialize(obj[key])}`).join(',')
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/applitools/sdk.coverage.tests/pullRequest?prid=8528488)
```suggestion
        const hooks = Object.keys(obj).map((key) => `${key}: ${serialize(obj[key])}`).join(',')
```